*** gnumake/make-4.1-original/function.c	2014-10-05 18:24:51.000000000 +0200
--- gnumake/make-4.1-patched/function.c	2015-02-03 16:59:39.614992654 +0100
***************
*** 20,29 ****
--- 20,31 ----
  #include "dep.h"
  #include "job.h"
  #include "commands.h"
  #include "debug.h"
  
+ #include "monmakeamoi.h"
+ 
  #ifdef _AMIGA
  #include "amiga.h"
  #endif
  
  
***************
*** 2304,2313 ****
--- 2306,2316 ----
    FT_ENTRY ("file",          1,  2,  1,  func_file),
  #ifdef EXPERIMENTAL
    FT_ENTRY ("eq",            2,  2,  1,  func_eq),
    FT_ENTRY ("not",           0,  1,  1,  func_not),
  #endif
+ MON_MAKE_A_MOI_FUNCTION_TABLE
  };
  
  #define FUNCTION_TABLE_ENTRIES (sizeof (function_table_init) / sizeof (struct function_table_entry))
  
  
*** gnumake/make-4.1-original/Makefile.am	2015-02-03 18:24:12.268146581 +0100
--- gnumake/make-4.1-patched/Makefile.am	2015-02-03 16:59:39.614992654 +0100
***************
*** 31,40 ****
--- 31,42 ----
  SUBDIRS =	glob config po doc $(MAYBE_W32)
  
  bin_PROGRAMS =	make
  include_HEADERS = gnumake.h
  
+ 
+ 
  if USE_CUSTOMS
    remote =	remote-cstms.c
  else
    remote =	remote-stub.c
  endif
***************
*** 49,70 ****
  
  noinst_HEADERS = commands.h dep.h filedef.h job.h makeint.h rule.h variable.h \
  		debug.h getopt.h gettext.h hash.h output.h
  
  make_LDADD =	@LIBOBJS@ @ALLOCA@ $(GLOBLIB) @GETLOADAVG_LIBS@ @LIBINTL@ \
! 		$(GUILE_LIBS)
  # Only process if target is MS-Windows
  if WINDOWSENV
      make_LDADD += $(W32LIB)
  endif
  
  man_MANS =	make.1
  
  DEFS =		-DLOCALEDIR=\"$(localedir)\" -DLIBDIR=\"$(libdir)\" -DINCLUDEDIR=\"$(includedir)\" @DEFS@
  
  AM_CPPFLAGS =	$(GLOBINC)
! AM_CFLAGS =	$(GUILE_CFLAGS)
  # Only process if target is MS-Windows
  if WINDOWSENV
      AM_CPPFLAGS +=	$(W32INC)
  endif
  
--- 51,72 ----
  
  noinst_HEADERS = commands.h dep.h filedef.h job.h makeint.h rule.h variable.h \
  		debug.h getopt.h gettext.h hash.h output.h
  
  make_LDADD =	@LIBOBJS@ @ALLOCA@ $(GLOBLIB) @GETLOADAVG_LIBS@ @LIBINTL@ \
! 		../../src/monmakeamoi.a $(GUILE_LIBS)
  # Only process if target is MS-Windows
  if WINDOWSENV
      make_LDADD += $(W32LIB)
  endif
  
  man_MANS =	make.1
  
  DEFS =		-DLOCALEDIR=\"$(localedir)\" -DLIBDIR=\"$(libdir)\" -DINCLUDEDIR=\"$(includedir)\" @DEFS@
  
  AM_CPPFLAGS =	$(GLOBINC)
! AM_CFLAGS =	-I../../src $(GUILE_CFLAGS)
  # Only process if target is MS-Windows
  if WINDOWSENV
      AM_CPPFLAGS +=	$(W32INC)
  endif
  
*** gnumake/make-4.1-original/main.c	2014-10-05 18:24:51.000000000 +0200
--- gnumake/make-4.1-patched/main.c	2015-02-03 18:11:14.804291347 +0100
***************
*** 214,223 ****
--- 214,227 ----
  
  /* List of makefiles given with -f switches.  */
  
  static struct stringlist *makefiles = 0;
  
+ /* Print debugging info (--debug).  */
+ 
+ static struct stringlist *stop_file = 0;
+ 
  /* Size of the stack when we started.  */
  
  #ifdef SET_STACK_SIZE
  struct rlimit stack_limit;
  #endif
***************
*** 381,390 ****
--- 385,396 ----
      N_("\
    -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE\n\
                                Consider FILE to be infinitely new.\n"),
      N_("\
    --warn-undefined-variables  Warn when an undefined variable is referenced.\n"),
+      N_("\
+   --stop-file=FILE            If the file exists, wait for the end of current jobs and exit.\n"),
      NULL
    };
  
  /* The table of command switches.
     Order matters here: this is the order MAKEFLAGS will be constructed.
***************
*** 433,442 ****
--- 439,449 ----
        &default_load_average, "load-average" },
  #endif
      { 'o', filename, &old_files, 0, 0, 0, 0, 0, "old-file" },
      { 'O', string, &output_sync_option, 1, 1, 0, "target", 0, "output-sync" },
      { 'W', filename, &new_files, 0, 0, 0, 0, 0, "what-if" },
+    
  
      /* These are long-style options.  */
      { CHAR_MAX+1, strlist, &db_flags, 1, 1, 0, "basic", 0, "debug" },
      { CHAR_MAX+2, string, &jobserver_fds, 1, 1, 0, 0, 0, "jobserver-fds" },
      { CHAR_MAX+3, flag, &trace_flag, 1, 1, 0, 0, 0, "trace" },
***************
*** 444,453 ****
--- 451,461 ----
        "no-print-directory" },
      { CHAR_MAX+5, flag, &warn_undefined_variables_flag, 1, 1, 0, 0, 0,
        "warn-undefined-variables" },
      { CHAR_MAX+6, strlist, &eval_strings, 1, 0, 0, 0, 0, "eval" },
      { CHAR_MAX+7, string, &sync_mutex, 1, 1, 0, 0, 0, "sync-mutex" },
+     { CHAR_MAX+100, filename, &stop_file, 0, 0, 0, 0, 0, "stop-file" },
      { 0, 0, 0, 0, 0, 0, 0, 0, 0 }
    };
  
  /* Secondary long names for options.  */
  
***************
*** 1394,1404 ****
       and we set the -p, -i and -e switches.  Doesn't seem quite right.  */
    decode_env_switches (STRING_SIZE_TUPLE ("MFLAGS"));
  #endif
  
    decode_switches (argc, (const char **)argv, 0);
! 
      /* Set a variable specifying whether stdout/stdin is hooked to a TTY.  */
  #ifdef HAVE_ISATTY
      if (isatty (fileno (stdout)))
        if (! lookup_variable (STRING_SIZE_TUPLE ("MAKE_TERMOUT")))
          define_variable_cname ("MAKE_TERMOUT", TTYNAME (fileno (stdout)),
--- 1402,1444 ----
       and we set the -p, -i and -e switches.  Doesn't seem quite right.  */
    decode_env_switches (STRING_SIZE_TUPLE ("MFLAGS"));
  #endif
  
    decode_switches (argc, (const char **)argv, 0);
!  
!  /* define a new variable MAKE_JOB_SLOTS containing the value of option '-j' 
!  
!  */
!  {
!  int processor=0;
!  int cores=0;
!  char tmpnjobs[100];
!  FILE* fcpuinfo;
!  sprintf(tmpnjobs,"%d",(job_slots<0?1:job_slots));
!  define_variable_cname ("MAKE_JOB_SLOTS", tmpnjobs,o_default, 0);
!  
!  fcpuinfo = fopen("/proc/cpuinfo","r");
!  if(fcpuinfo!=NULL)
!     {
!     while(fgets(tmpnjobs,100,fcpuinfo)!=0)
!         {
!         if(strncmp(tmpnjobs,"processor",9)==0) processor++;
!         else if(strncmp(tmpnjobs,"cpu cores",9)==0)
!             {
!             int ncores=0;
!             char* colon=strchr(tmpnjobs,':');
!             if(colon!=0) ncores=atoi(colon+1);
!             cores=(cores==0 || cores < ncores ? ncores : cores);
!             }
!         }
!     fclose(fcpuinfo);
!     }
!  sprintf(tmpnjobs,"%d",(processor==0?1:processor));
!  define_variable_cname ("MACHINE_PROCESSORS", tmpnjobs,o_default, 0);
!  sprintf(tmpnjobs,"%d",(cores==0?1:cores));
!  define_variable_cname ("MACHINE_CORES", tmpnjobs,o_default, 0);
!  }
      /* Set a variable specifying whether stdout/stdin is hooked to a TTY.  */
  #ifdef HAVE_ISATTY
      if (isatty (fileno (stdout)))
        if (! lookup_variable (STRING_SIZE_TUPLE ("MAKE_TERMOUT")))
          define_variable_cname ("MAKE_TERMOUT", TTYNAME (fileno (stdout)),
***************
*** 1502,1511 ****
--- 1542,1553 ----
      argv[0] = xstrdup (concat (3, current_directory, "/", argv[0]));
  #endif /* !__MSDOS__ */
  #endif /* WINDOWS32 */
  #endif
  
+ 
+  
    /* We may move, but until we do, here we are.  */
    starting_directory = current_directory;
  
  #ifdef MAKE_JOBSERVER
    /* If the jobserver-fds option is seen, make sure that -j is reasonable.
***************
*** 1541,1551 ****
--- 1583,1595 ----
           jobserver.  If !job_slots and we don't have a pipe, we can start
           infinite jobs.  If we see both a pipe and job_slots >0 that means the
           user set -j explicitly.  This is broken; in this case obey the user
           (ignore the jobserver pipe for this make) but print a message.
           If we've restarted, we already printed this the first time.  */
+    
  
+    
        if (job_slots > 0)
          {
            if (! restarts)
              O (error, NILF,
                 _("warning: -jN forced in submake: disabling jobserver mode."));
***************
*** 1588,1598 ****
            free (jobserver_fds);
            jobserver_fds = 0;
          }
      }
  #endif
! 
    /* The extra indirection through $(MAKE_COMMAND) is done
       for hysterical raisins.  */
  #ifdef VMS
    define_variable_cname("MAKE_COMMAND", vms_command(argv[0]), o_default, 0);
  #else
--- 1632,1642 ----
            free (jobserver_fds);
            jobserver_fds = 0;
          }
      }
  #endif
!    
    /* The extra indirection through $(MAKE_COMMAND) is done
       for hysterical raisins.  */
  #ifdef VMS
    define_variable_cname("MAKE_COMMAND", vms_command(argv[0]), o_default, 0);
  #else
***************
*** 1961,1970 ****
--- 2005,2015 ----
      /* If we've disabled builtin variables, get rid of them.  */
      if (no_builtin_variables_flag && ! old_builtin_variables_flag)
        undefine_default_variables ();
    }
  
+  
  #if defined (__MSDOS__) || defined (__EMX__) || defined (VMS)
    if (job_slots != 1
  # ifdef __EMX__
        && _osmode != OS2_MODE /* turn off -j if we are in DOS mode */
  # endif
***************
*** 3451,3455 ****
--- 3496,3513 ----
          }
      }
  
    exit (status);
  }
+ 
+ 
+ int is_stop_file_created()
+ 	{
+ 	FILE* f;
+ 	if( stop_file == 0 || stop_file->list==0) return 0;
+ 	f=fopen(stop_file->list[0],"r");
+ 	if(f==NULL) return 0;
+ 	fclose(f);
+ 	OS(error, NILF, _("Warning: Stop-file '%s' detected. Waiting for end of current jobs."),  stop_file->list[0]);
+ 	return 1;
+ 	}
+ 
